### PR TITLE
comment realtime

### PR DESCRIPTION
**Pull request by NgengChhengkim**

### CODE REVIEW

File: app/jobs/comment_job.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class CommentJob < ApplicationJob
+  queue_as :default
+
+  def perform comment
+    activity = comment.activity
+    trackable = activity.trackable
+    notifications = trackable.notifications
+    receiver_id = trackable.user_id
+    notification = notifications.find_or_create_by notify_type: Settings.notifications.notify_types.comment
```
Comment (yongsokheng) [inline app/jobs/comment_job.rb]: notifications.find_or_create_by ... ?
run dung ko?

File: app/services/reaction_broad_cast_service.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  attr_accessor :channel, :comment, :activity_id
+
+  def initialize channel, comment, activity_id
+    @channel = channel
+    @comment = comment
+    @activity_id = activity_id
+  end
+
+  def broadcast
+    ActionCable.server.broadcast channel, content: render_content, activity_id: activity_id
```
Comment (yongsokheng) [inline app/services/reaction_broad_cast_service.rb:11]: activity_id or @activity_id ?

File: app/assets/javascripts/channels/reactions.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  },
+
+  disconnected: function() {
+  },
+
+  received: function(data) {
+    var activity = $(".new-feeds-activity-" + data.activity_id);
+    var comment_count = activity.find(".comment a");
+
+    var count = (comment_count.data().count += 1);
```
Comment (yongsokheng) [inline app/assets/javascripts/channels/reactions.js:12]: comment_count.data().count ?

File: app/channels/reactions_channel.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class ReactionsChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from "reaction_channel"
```
Comment (yongsokheng) [inline app/channels/reactions_channel.rb:3]: channel ot mean chap id te?

Comment (NgengChhengkim) [reply app/channels/reactions_channel.rb:3]: dak id ey? kreng chanel neng nak na co tor toul ban dea oy te see activity?

Comment (yongsokheng) [reply app/channels/reactions_channel.rb:3]: OK, kit chrolom :D

File: app/assets/javascripts/channels/reactions.js (hunk 3 above)
Comment (NgengChhengkim) [reply app/assets/javascripts/channels/reactions.js:12]: cong them 1 hoy ban gan oy count. viet cheng data-count knong html co kern dea. lerk kroy yok mok teat ban trov. kom ey hao comment text count yok mok hoy. te chop ().

File: app/services/reaction_broad_cast_service.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/services/reaction_broad_cast_service.rb:11]: hao activity_id co chenh dea. te kit tha @activity_id trov cheang. knong notification_chanel u hao cheng dea.

Comment (yongsokheng) [reply app/services/reaction_broad_cast_service.rb:11]: kit oy cbas tov doch trov ta. 

```
def activity_id
 @activity_id 
end
```

cheng bao activity_id rur @activity_id doch khnea ta

File: app/jobs/comment_job.rb (hunk 1 above)
Comment (NgengChhengkim) [reply app/jobs/comment_job.rb]: kon leng neng khos ey ke?

File: app/assets/javascripts/channels/reactions.js (hunk 3 above)
Comment (yongsokheng) [reply app/assets/javascripts/channels/reactions.js:12]: comment_count.data() neng ey ke?

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -4,7 +4,8 @@ class CommentsController < ApplicationController
   def index
     @activity = Activity.find params[:comment][:activity_id]
     @comments = @activity.comments.preload(:user).order(id: :desc)
-      .page(params[:page]).per Settings.comments.limit
+      .page(params[:page]).per(Settings.comments.limit)
+      .padding params[:comment][:new_comment_count].to_i
```
Comment (yongsokheng) [inline app/controllers/comments_controller.rb]: new_comment_count cham bach dak knong params :comment de?

File: app/jobs/comment_job.rb (hunk 1 above)
Comment (yongsokheng) [reply app/jobs/comment_job.rb]: notifications yok bos post/share post type comment

File: app/services/reaction_broad_cast_service.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/services/reaction_broad_cast_service.rb:11]: cheng dak ey? :hear_no_evil:

File: app/assets/javascripts/channels/reactions.js (hunk 3 above)
Comment (NgengChhengkim) [reply app/assets/javascripts/channels/reactions.js:12]: `comment_count.data().count == comment_count.data("count")`.  viet cheng kran te kley
cong them 1 hoy set gia tri thmey oy `comment_count.data("count")`

Comment (yongsokheng) [reply app/assets/javascripts/channels/reactions.js:12]: ot del khernh :smile:

File: app/services/reaction_broad_cast_service.rb (hunk 2 above)
Comment (yongsokheng) [reply app/services/reaction_broad_cast_service.rb:11]: dak cheng hoy

File: app/assets/javascripts/new_feeds.js [hunk 6]
```diff
@@ -31,28 +31,39 @@ $(document).on("turbolinks:load ajaxComplete", function() {
...
+      new_feeds.find(".comments-content").show();
       $(this).unbind("click").removeClass("show-comments");
-      $(".comments-spinner").show();
-      $.get("/comments", {comment: {activity_id: activity_id}});
+      var new_comment_count = new_feeds.find(".comment-form").data("new-comment-count");
+
+      if (comment_count > 0) {
+        var activity_id = new_feeds.find(".comments").data("activity-id");
+        new_feeds.find(".comments-spinner").show();
+        $.get("/comments", {comment: {activity_id: activity_id, new_comment_count: new_comment_count}});
```
Comment (yongsokheng) [inline app/assets/javascripts/new_feeds.js]: params `new_comment_count` nov knong comment ? khernh controller dak `params[:new_comment_count]`

**Comment by yongsokheng** [comment]
OK

**Pull request merged**